### PR TITLE
Update version and help strings for Zcash

### LINKED ISSUE #938: Update ``clientversion.h`` to support our ``.z<N>`` suffixes on versions.
The goal is for --version to show our release versions (which should be identical to the associated git tag).

### PR DESCRIPTION
**Pull request by str4d**
Closes #938.

### CODE REVIEW

**Comment by ebfull** [comment]
ACK but nathan would know better.

File: configure.ac [hunk 1]
```diff
@@ -4,9 +4,10 @@ define(_CLIENT_VERSION_MAJOR, 0)
 define(_CLIENT_VERSION_MINOR, 11)
 define(_CLIENT_VERSION_REVISION, 2)
 define(_CLIENT_VERSION_BUILD, 0)
+define(_CLIENT_VERSION_ZCASH, 3)
 define(_CLIENT_VERSION_IS_RELEASE, true)
 define(_COPYRIGHT_YEAR, 2015)
-AC_INIT([Bitcoin Core],[_CLIENT_VERSION_MAJOR._CLIENT_VERSION_MINOR._CLIENT_VERSION_REVISION],[https://github.com/bitcoin/bitcoin/issues],[bitcoin])
+AC_INIT([Bitcoin Core],[_CLIENT_VERSION_MAJOR._CLIENT_VERSION_MINOR._CLIENT_VERSION_REVISION.z_CLIENT_VERSION_ZCASH],[https://github.com/bitcoin/bitcoin/issues],[bitcoin])
```
Comment (nathan-at-least) [inline configure.ac]: I don't know what context the ``Bitcoin Core`` label is used in, but let's go ahead and replace it with ``Zcash`` and if anything breaks from that, fix it in follow up PRs.

File: configure.ac [hunk 2]
```diff
@@ -7,7 +7,7 @@ define(_CLIENT_VERSION_BUILD, 0)
 define(_CLIENT_VERSION_ZCASH, 3)
 define(_CLIENT_VERSION_IS_RELEASE, true)
 define(_COPYRIGHT_YEAR, 2015)
-AC_INIT([Bitcoin Core],[_CLIENT_VERSION_MAJOR._CLIENT_VERSION_MINOR._CLIENT_VERSION_REVISION.z_CLIENT_VERSION_ZCASH],[https://github.com/bitcoin/bitcoin/issues],[bitcoin])
+AC_INIT([Zcash Core],[_CLIENT_VERSION_MAJOR._CLIENT_VERSION_MINOR._CLIENT_VERSION_REVISION.z_CLIENT_VERSION_ZCASH],[https://github.com/zcash/zcash/issues],[zcash])
```
Comment (nathan-at-least) [inline configure.ac]: (Prior commit's comment superceded; I'll go update it.) Let's drop "Core".

File: src/bitcoin-cli.cpp [hunk 3]
```diff
@@ -63,12 +63,12 @@ static bool AppInitRPC(int argc, char* argv[])
     //
     ParseParameters(argc, argv);
     if (argc<2 || mapArgs.count("-?") || mapArgs.count("-h") || mapArgs.count("-help") || mapArgs.count("-version")) {
-        std::string strUsage = _("Bitcoin Core RPC client version") + " " + FormatFullVersion() + "\n";
+        std::string strUsage = _("Zcash Core RPC client version") + " " + FormatFullVersion() + "\n";
```
Comment (nathan-at-least) [inline src/bitcoin-cli.cpp]: Drop "Core".

File: src/bitcoind.cpp [hunk 4]
```diff
@@ -69,7 +69,7 @@ bool AppInit(int argc, char* argv[])
     // Process help and version before taking care about datadir
     if (mapArgs.count("-?") || mapArgs.count("-h") ||  mapArgs.count("-help") || mapArgs.count("-version"))
     {
-        std::string strUsage = _("Bitcoin Core Daemon") + " " + _("version") + " " + FormatFullVersion() + "\n";
+        std::string strUsage = _("Zcash Core Daemon") + " " + _("version") + " " + FormatFullVersion() + "\n";
```
Comment (nathan-at-least) [inline src/bitcoind.cpp]: Drop "Core".

File: src/init.cpp [hunk 5]
```diff
@@ -448,6 +448,7 @@ std::string HelpMessage(HelpMessageMode mode)
 std::string LicenseInfo()
 {
     return FormatParagraph(strprintf(_("Copyright (C) 2009-%i The Bitcoin Core Developers"), COPYRIGHT_YEAR)) + "\n" +
+           FormatParagraph(strprintf(_("Copyright (C) 2015-%i The Zcash Core Developers"), COPYRIGHT_YEAR)) + "\n" +
```
Comment (nathan-at-least) [inline src/init.cpp]: Drop "Core".

**Comment by nathan-at-least** [comment]
Awesome! I'm happy to see this. The only change I want to see is that we drop "Core". The project's just "Zcash".

**Comment by str4d** [comment]
@nathan-at-least done!

**Comment by nathan-at-least** [comment]
ACK.

@zkbot r+

File: src/bitcoind.cpp [hunk 6]
```diff
@@ -110,19 +110,19 @@ bool AppInit(int argc, char* argv[])
         // Command-line RPC
         bool fCommandLine = false;
         for (int i = 1; i < argc; i++)
-            if (!IsSwitchChar(argv[i][0]) && !boost::algorithm::istarts_with(argv[i], "bitcoin:"))
+            if (!IsSwitchChar(argv[i][0]) && !boost::algorithm::istarts_with(argv[i], "zcash:"))
                 fCommandLine = true;
 
         if (fCommandLine)
         {
-            fprintf(stderr, "Error: There is no RPC client functionality in bitcoind anymore. Use the bitcoin-cli utility instead.\n");
```
Comment (daira) [inline src/bitcoind.cpp]: Delete " anymore".

File: src/bitcoind.cpp [hunk 7]
```diff
@@ -110,19 +110,19 @@ bool AppInit(int argc, char* argv[])
...
+            fprintf(stderr, "Error: There is no RPC client functionality in zcashd anymore. Use the zcash-cli utility instead.\n");
```
Comment (daira) [inline src/bitcoind.cpp]: Delete " anymore".

**Comment by ebfull** [comment]
@zkbot r-

@str4d can you fix the minor nit daira brought up after this was ack'd by nathan? Gives buildbot time to calm down.

**Comment by ebfull** [comment]
@zkbot r-

**Comment by str4d** [comment]
Fixed!

**Comment by ebfull** [comment]
Please work.

@zkbot r+ b3e712a4d22930b0cc62a430814345aa9f58f910

**Pull request merged**